### PR TITLE
[WIP]: add ability to upvote players

### PR DESCRIPTION
**Pull request by SergeyKhval**
This PR adds ability to upvote a player. This will be used from Twitch extension, so I'll need to validate their JWT before allowing to upvote

### CODE REVIEW

**Comment by modmoto** [comment]
But now we can upvote mutliple times, right? Would that not be easily exploitable? The identity service hands out jwts now. So if you want to validate the person upvoting, this should be doable. I am moving the backend to use the jwts soon, so maybe we reuse that there.

File: W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs [hunk 1]
```diff
@@ -104,5 +104,13 @@ public async Task<IActionResult> MigrateCountry()
 
             return Ok();
         }
+
+        [HttpPost("{battleTag}/upvote")]
+        public async Task<int> UpvotePlayer(string battleTag)
+        {
+            var result = await _personalSettingsRepository.Upvote(battleTag);
```
Comment (modmoto) [inline W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs:111]: like mentioned, multiple upvotes should not be possible

**Comment by SergeyKhval** [comment]
Yes, users should be able to upvote multiple times, because I'm going to integrate this feature with Twitch bits. So they will basically spend bits to buy upvotes.

File: W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs (hunk 1 above)
Comment (SergeyKhval) [reply W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs:111]: It should be possible. The flow is as follows:

1. User clicks on upvote button in Twitch Ext
2. They receive a confirmation from twitch that they enough bits / purchase bits
3. If the purchase is completed we send a request to our backend to upvote

Comment (modmoto) [reply W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs:111]: But if I am smart, I just use this endpoint independantly from twitch and give me a million upvotes? Can we add a secret that is needed for the twitch extension? Or are the extension secrets open like on a website? Or maybe do this via cors, so only twitch can acces that endpoint. Dont know. But we would need some sort of validation within that handler.

Comment (SergeyKhval) [reply W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs:111]: That's exactly the idea. Each request from Twitch extension will have a signed JWT. On the backend I will store Twitch secret to verify this JWT. Once verified, I can proceed with upvote

Comment (modmoto) [reply W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs:111]: Ok perfect, but then we wait until the jwt stuff is also included? I dont think people will really go through the code and search for this endpoint, but as it is connected to money, we should wait until the endpoint is secured. Or featuretoggle it.

Comment (SergeyKhval) [reply W3ChampionsStatisticService/PersonalSettings/PersonalSettingsController.cs:111]: Yes, no need to merge now. I'll mark this as a WIP

**Pull request closed without merging**